osf/models/base.py

```diff
@@ -208,15 +208,39 @@ class Guid(BaseModel):
     def __repr__(self):
         return f'<id:{self._id}, referent:({self.referent.__repr__()})>'
 
+    @classmethod
+    def split_guid(cls, guid):
+        if not guid:
+            return None, None
+        guid_parts = guid.split(VersionedGuidMixin.GUID_VERSION_DELIMITER)
+        base_guid = guid_parts[0]
+        version = guid_parts[1] if len(guid_parts) > 1 else None
+        return base_guid, version
+
     # Override load in order to load by GUID
     @classmethod
     def load(cls, data, select_for_update=False):
+        base_guid, version = cls.split_guid(data)
         try:
-            return cls.objects.get(_id=data) if not select_for_update else cls.objects.filter(
-                _id=data).select_for_update().get()
+            return cls.objects.get(_id=base_guid) if not select_for_update else cls.objects.filter(
+                _id=base_guid).select_for_update().get()
         except cls.DoesNotExist:
             return None
 
+    @classmethod
+    def load_referent(cls, guid):
+        base_guid, version = cls.split_guid(guid)
+
+        base_guid_obj = cls.load(base_guid)
+        if version and base_guid_obj.is_versioned:
+            versioned_obj_qs = base_guid_obj.versions.filter(version=version)
+            if not versioned_obj_qs.exists():
+                return None
+
+            return versioned_obj_qs.first().referent
```

Comment:
(New ticket) Add a new uniqueness constraint for the through table. With that, we can easily do

```python
try:
    return base_guid_obj.versions.get(version=version).referent
except GuidVersionsThrough.DoesNotExists:
    return None
```